src/optimize/mod.rs

```diff
@@ -5,8 +5,7 @@ mod stack2reg;
 
 pub(crate) fn optimize_function<'tcx>(
     tcx: TyCtxt<'tcx>,
-    #[cfg_attr(not(debug_assertions), allow(unused_variables))]
-    instance: Instance<'tcx>,
+    #[cfg_attr(not(debug_assertions), allow(unused_variables))] instance: Instance<'tcx>,
```

Comment:
```suggestion
    instance: Instance<'tcx>,
```

It has been used in all cases for a while.